src/lib.rs

```diff
@@ -73,15 +73,16 @@ mod error;
 use std::slice;
 use std::mem::{align_of, forget};
 
+pub mod guard;
```

Comment:
Import lines in shortest-to-longest, import elements longest-to-shortest, prioritised as modules then types then functions, I thought this was standard procedure?